lib/evmone/instructions.hpp

```diff
@@ -899,6 +899,68 @@ inline code_iterator swapn(StackTop stack, ExecutionState& state, code_iterator
     return pos + 2;
 }
 
+inline Result dataload(StackTop stack, int64_t gas_left, ExecutionState& state) noexcept
+{
+    auto& index = stack.top();
+
+    if (state.data.size() < 32 || (state.data.size() - 32) < index)
+        return {EVMC_INVALID_MEMORY_ACCESS, gas_left};  // TODO better error
```

Comment:
What would be a better error? Specific DATALOAD offset error?

Reply:
I was thinking it could be something like `EVMC_INVALID_DATA_ACCESS`, but not sure anymore if it's important, will just remove TODO for now.

Reply:
Right, I think that could make sense, but even `EVMC_INVALID_MEMORY_ACCESS` is okay probably. I'm fine keeping a comment mentioning "TODO: Introduce EVMC_INVALID_DATA_ACCESS".

Reply:
Added comments.